spec/workers/webhook_worker_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe WebhookWorker do
+  describe "perform" do
+    let(:report) { { some: "json" } }
+    let(:webhook_uri) { "http://webhooks-rule.org/webhook" }
+    let(:secret_key) { nil }
+
+    context "with a secret key" do
+      let(:secret_key) { "this is a secret key" }
+
+      # http://www.freeformatter.com/hmac-generator.html
+      let(:expected_signature) { "d44d805a16b90922edb48f14ff4e858a5d3e39bd" }
```

Comment:
Since we can generate the signature with one line it might be better to just generate it here than using the external site to do it.